test/test.js

```diff
@@ -2,10 +2,23 @@
  * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
  */
 const bedrock = require('bedrock');
+const sinon = require('sinon');
+const brPassport = require('bedrock-passport');
+/* passportStub is defined globally here because we need to stub
```

Comment:
If you want to try stubbing `bedrock-passport`'s `authenticate` method that might work, but it sounds like you have investigated this and know what you are doing so I 100% support this method especially if it saves on time.